src/player/C4Player.cpp

```diff
@@ -1783,3 +1773,100 @@ void C4Player::SetSoundModifier(C4PropList *new_modifier)
+			case P_Type:                throw new C4AulExecError("player: Type is read-only");
+			case P_CrewSkin:            throw new C4AulExecError("player: CrewSkin is read-only");
+			case P_ExtraID:             throw new C4AulExecError("player: ExtraID is read-only");
+			case P_InitialScore:        throw new C4AulExecError("player: InitialScore is read-only");
+			case P_ZoomLimit_MaxHeight: throw new C4AulExecError("player: ZoomLimit is read-only");
+			case P_ZoomLimit_MaxValue:  throw new C4AulExecError("player: ZoomLimit is read-only");
+			case P_ZoomLimit_MaxWidth:  throw new C4AulExecError("player: ZoomLimit is read-only");
+			case P_ZoomLimit_MinHeight: throw new C4AulExecError("player: ZoomLimit is read-only");
+			case P_ZoomLimit_MinValue:  throw new C4AulExecError("player: ZoomLimit is read-only");
+			case P_ZoomLimit_MinWidth:  throw new C4AulExecError("player: ZoomLimit is read-only");
+		}
+	}
+}
+
+void C4Player::SetPropertyByS(C4String * k, const C4Value & to)
+{
+	ProtectReadonlyProperty(k);
+	if (k >= &Strings.P[0] && k < &Strings.P[P_LAST])
+	{
+		switch(k - &Strings.P[0])
+		{
+			case P_Score: CurrentScore = to.getInt(); return;
+		}
+	}
+	C4PropList::SetPropertyByS(k, to);
+}
+
+void C4Player::ResetProperty(C4String * k)
+{
+	ProtectReadonlyProperty(k);
+	if (k >= &Strings.P[0] && k < &Strings.P[P_LAST])
+	{
+		switch(k - &Strings.P[0])
+		{
+			case P_Score: CurrentScore = InitialScore; return;
+		}
+	}
+	C4PropList::ResetProperty(k);
+}
+
+bool C4Player::GetPropertyByS(const C4String *k, C4Value *pResult) const
+{
+	if (k >= &Strings.P[0] && k < &Strings.P[P_LAST])
+	{
+		switch(k - &Strings.P[0])
+		{
+			case P_ID:           *pResult = C4VInt(ID);               return true;
+			case P_Name:         *pResult = C4VString(Name);          return true;
+			case P_Type:         *pResult = C4VInt(GetType());        return true;
+			case P_CrewSkin:     *pResult = C4VInt(PrefClonkSkin);    return true;
+			case P_InitialScore: *pResult = C4VInt(InitialScore);     return true;
+			case P_Score:        *pResult = C4VInt(CurrentScore);     return true;
+			case P_ZoomLimit_MaxWidth:  *pResult = C4VInt((ZoomLimitMaxWdt || ZoomLimitMaxHgt) ? ZoomLimitMaxWdt : C4VP_DefMaxViewRangeX); return true;
+	        case P_ZoomLimit_MaxHeight: *pResult = C4VInt(ZoomLimitMaxHgt); return true;
+	        case P_ZoomLimit_MaxValue:  *pResult = C4VInt(fixtoi(ZoomLimitMaxVal, 100)); return true;
+	        case P_ZoomLimit_MinWidth:  *pResult = C4VInt((ZoomLimitMinWdt || ZoomLimitMinHgt) ? ZoomLimitMinWdt : C4VP_DefMinViewRangeX); return true;
+	        case P_ZoomLimit_MinHeight: *pResult = C4VInt(ZoomLimitMinHgt); return true;
+	        case P_ZoomLimit_MinValue:  *pResult = C4VInt(fixtoi(ZoomLimitMinVal, 100)); return true;
+			case P_ExtraID: // Gives me a error: jump to case label [-fpermissive] caused by the line *info if I use this at any of the earlier cases..
```

Comment:
Ah, yes, take a look here please, that part gave me weird compile errors if I move it before another case, so I guess it could lead to other errors.